src/warc2zim/url_rewriting.py

```diff
@@ -63,8 +63,8 @@
 
 FUZZY_RULES = [
     {
-        "pattern": r".*googlevideo.com/(videoplayback\?).*((?<=[?&])id=[^&]+).*",
-        "replace": r"youtube.fuzzy.replayweb.page/\1\2",
+        "pattern": r".*googlevideo.com/(videoplayback(?=\?)).*[?&](id=[^&]+).*",
```

Comment:
Maybe I've misunderstood, but how is this different from just doing:

`(videoplayback\?).*[?&]` (etc.)?

You don't need a lookahead, ISTM, if you simply want to match a question mark that will always be there. The `.*` after the lookahead will consume the matching `?` anyway. Only reason to use a lookahead is when you don't want to consume any characters but you want to know that there is something ahead.

Reply:
Or, if you don't want to match the question mark in the first parentheses, which looking at it I presume you don't, then put `\?` outside the first parens.

Reply:
> The `.*` after the lookahead will consume the matching `?`

It will consume **until** the `?` but it will not consume it.

And we don't want to consume it (first) to have the `[?&]` consume it.
With `(videoplayback\?)` we consume the `?` and so `[?&]`  will never match on `googlevideo.com/videoplayback?id=1234`

Reply:
Original pattern consumed the question mark inside the first parentheses...

Reply:
OK, I've understood it now! Thanks for the explanations!